### ISSUE TITLE
Time_Sent value in header has incorrect value 

### ISSUE THREAD

**Issue by WojcikMike** [Particular/NServiceBus#3508]
While running my sample application (https://github.com/WojcikMike/PubSubSample) and validating results in ServiceInsight I noticed wierd behavior for 1 of the cases:
When one of the  Subscribers are throwing exception the time_sent value in header has bad value in it:

"time_sent": "2016-02-24T10:37:56.12074Z",
"processed_at":"2016-02-24T10:37:56.115736Z",

So the time sent in above case is greater than processed_at. What is more the value time_sent is different for EventMessage processed by Subscriber 1 and Subscriber 2:
"time_sent":"2016-02-24T10:37:52.138186Z", - Subscriber 1
"time_sent":"2016-02-24T10:37:56.12074Z", - Subscriber 2

Comparing to v5 this is differnet behavior in a way that: time_sent has lower value than processed_at and time_sent in above case is equal for Subscriber1 and Subscriber2

**Comment by andreasohlund** [comment]
Since v6 no longer sends the same message multiple times but instead a clone we need to decide what "time sent" really means. When do we stamp that time?

When user calls bus.Send/Publish? (if yes how do we deal with control messages )

Do we stamp it when the infrastructure sends it? (given that we have batching in v6 it can be quite a difference)

processed_at being > time_sent seems like a bug to me!

Could it be related to https://github.com/Particular/NServiceBus/issues/3494 ?

**Comment by WojcikMike** [comment]
@andreasohlund yeah it's conceptual discussion. It seems that at the moment time sent is set in here:
https://github.com/Particular/NServiceBus/blob/b97f7ba496028394c003b0537fb159076f1d4135/src/NServiceBus.Core/Transports/OutgoingMessage.cs#L24

So whenever we are creating OutgoingMessage the time_sent is overriden, which I believe also means that while doing audit it actually stores time_sent of audited message, so the actual time_sent of original message is not available. Unless I understood it the wrong way.

I am almost sure that by changing the value of TimeSent we need to adjust ServiceInsight as I think it rely on that value in some way.

**Comment by WojcikMike** [comment]
I removed in progress label as we need to agree what is expected behavior for v6 and then fix it.

**Comment by andreasohlund** [comment]
>which I believe also means that while doing audit it actually stores time_sent of audited message, so the actual time_sent of original message is not available. Unless I understood it the wrong way.

Yea, this is wrong, and would cause the processedAt to be less than sent_time.

Let's get on a call tomorrow and see how we can find a way to fix this

**Comment by WojcikMike** [comment]
After discussing with Andreas the attack plan is to fix the time_sent to contains value as close to ```bus.Send/Publish``` as possible.

**Comment by WojcikMike** [comment]
While discussing with @andreasohlund @danielmarbach @timbussmann it was decided that we will introduce less elegant solution by making sure that time_sent and nservicebusverion is not overriden during audits. 

We all agreed that more elegant solution is desirable but it requires more discussions and broader changes. New issue will be created by @danielmarbach

**Comment by danielmarbach** [comment]
I think we assigned @tmasternak to that. See https://github.com/Particular/NServiceBus/issues/3532

**Comment by tmasternak** [comment]
@danielmarbach I think that #3532 is about TM and this one is about headers in audit messages.
